packages/widgets/entity-dms/src/components/EntityDms/EntityDms.js

```diff
@@ -0,0 +1,55 @@
+import PropTypes from 'prop-types'
+import DocsBrowserApp from 'tocco-docs-browser/src/main'
+import {LoadMask} from 'tocco-ui'
+import {appContext as appContextPropType} from 'tocco-util'
+
+const EntityDms = ({folderKey, listLimit, backendUrl, businessUnit, appContext}) => {
+  const getListFormName = (parent, keys) => {
+    if (parent !== null) {
+      return 'DmsPublic_docs_list_item'
+    } else if (keys !== null) {
+      return 'DmsPublic_root_docs_list_item_specific'
+    } else {
+      return 'DmsPublic_root_docs_list_item'
+    }
+  }
+
+  const rootNodes = []
+
+  if (folderKey) {
+    rootNodes.push({
```

Comment:
nitpick: I personally do not like functions that mutate objects/arrays and I try to write code that only works with immutable objects

```
const rootNodes = folderKey ? [{...}] : []
```